src/test/java/io/jenkins/plugins/checks/github/CheckRunGHEventSubscriberTest.java

```diff
@@ -91,6 +91,17 @@ void shouldIgnoreHeadBranchMissingFromPayload() throws IOException {
         assertThat(loggerRule.getMessages().get(0)).contains("Received rerun request through GitHub checks API.");
     }
 
+    @Test
+    void shouldThrowExceptionWhenHeadBranchMissingFromPayload() throws IOException {
+        assertThatThrownBy(
+            () -> {
+                new CheckRunGHEventSubscriber(mock(JenkinsFacade.class), mock(SCMFacade.class))
+                  .onEvent(createEventWithRerunRequest(RERUN_REQUEST_JSON_FOR_PR_MISSING_CHECKSUITE_HEAD_BRANCH));
+            })
+                .isInstanceOf(IllegalStateException.class)
+                .hasMessageContaining("Could not parse check run event:");
+    }
+
```

Comment:
Bad merge by the looks of it, the behaviour was changed in https://github.com/jenkinsci/github-checks-plugin/commit/f35f09a146fb2f75c4f2bf3f5d6ec0f02d00e06a
```suggestion
```